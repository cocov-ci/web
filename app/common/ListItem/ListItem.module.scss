@import "styles/variables.module";

.listBox {
  .listItem {
    padding: 0;
    cursor: pointer;

    .content {
      display: flex;
      justify-content: space-between;
    }

    .info {
      padding: 28px;
      min-width: 0;
      z-index: 1;
      position: relative;

      .title {
        font-variation-settings: "wght" 600;
        margin-bottom: 5px;
        white-space: nowrap;
        overflow: hidden;
        text-overflow: ellipsis;
      }

      .description {
        font-variation-settings: "wght" 400;

        &.disabled {
          color: $colorGrayMedium;
        }
      }
    }

    &.loading {
      cursor: wait;
    }

    .skeleton {
      line-height: 0;
      border-radius: 0;
    }

    .stats {
      display: flex;

      > div {
        border-left: 1px solid $colorGray;
      }
    }
  }
}
