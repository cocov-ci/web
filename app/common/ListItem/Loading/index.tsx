import classNames from 'classnames'

import Box from 'app/common/Box'
import styles from 'app/common/ListItem/ListItem.module.scss'
import Loading from 'app/common/Loading'
import randomBetween from 'utils/randomBetween'

const LoadingListItem = () => {
  return (
    <Box className={classNames(styles.listItem, styles.loading)}>
      <div className={styles.info}>
        <Loading
          alignment="left"
          height="30px"
          type="skeleton"
          width={`${randomBetween(35, 465)}px`}
        />
        <Loading
          alignment="left"
          height="19px"
          type="skeleton"
          width={`${randomBetween(35, 200)}px`}
        />
      </div>
      <div className={styles.stats}>
        <Loading
          alignment="left"
          className={styles.skeleton}
          height="116px"
          type="skeleton"
          width="173px"
        />
        <Loading
          alignment="left"
          className={styles.skeleton}
          height="116px"
          type="skeleton"
          width="173px"
        />
      </div>
    </Box>
  )
}

export default LoadingListItem
