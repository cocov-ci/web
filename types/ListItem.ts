import { StatsProps } from './Stats'

export interface ListItemComponentProps {
  title: string
  description?: string
  stats: StatsProps
}
