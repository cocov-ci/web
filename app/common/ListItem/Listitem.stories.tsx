import type { Meta, StoryObj } from '@storybook/react'

import ListItem from './index'

const meta: Meta<typeof ListItem> = {
  title: 'Common/ListItem',
  component: ListItem,
  tags: ['autodocs'],
}

export default meta

type Story = StoryObj<typeof ListItem>

export const Default: Story = {
  args: {
    title:
      'Titlefsdkjfjksdfljskdlkfjlsdjflksjdlfkjsldkfjlskdjflkjsdlkfjsldkfjlkdshere',
    description: 'Description here...',
    stats: {
      issues: {
        data: new Array(31).fill(0).map(() => Math.ceil(Math.random() * 10000)),
        value: Math.ceil(Math.random() * 1000),
      },
      coverage: {
        data: new Array(31).fill(0).map(() => Math.ceil(Math.random() * 10000)),
        value: Math.ceil(Math.random() * 1000),
      },
    },
  },
}
