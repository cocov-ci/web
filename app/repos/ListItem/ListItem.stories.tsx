import type { Meta, StoryObj } from '@storybook/react'

import ListItem from './index'

const meta: Meta<typeof ListItem> = {
  title: 'Repositories/ListItem',
  component: ListItem,
  tags: ['autodocs'],
}

export default meta

type Story = StoryObj<typeof ListItem>

export const Default: Story = {
  args: {
    name: 'jps',
    description: 'Josie Platform Server',
  },
}
