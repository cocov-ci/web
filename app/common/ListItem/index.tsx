import classNames from 'classnames'

import Box from 'app/common/Box'
import Stats from 'app/common/Stats'
import Text from 'app/common/Text'
import { ListItemComponentProps } from 'types/ListItem'

import styles from './ListItem.module.scss'

const ListItem = ({ title, description, stats }: ListItemComponentProps) => {
  return (
    <Box className={styles.listItem}>
      <span
        className={classNames(styles.backgroundText, {
          [styles.small]: title.length > 4,
        })}
      >
        {title}
      </span>
      <div className={styles.info}>
        <Text className={styles.title} variant="title">
          {title}
        </Text>
        <Text
          className={classNames(styles.description, {
            [styles.disabled]: !description,
          })}
          variant="description"
        >
          {description || 'No description'}
        </Text>
      </div>
      <div className={styles.stats}>
        <Stats data={stats.issues} type="issues" />
        <Stats data={stats.coverage} type="coverage" />
      </div>
    </Box>
  )
}

export default ListItem
