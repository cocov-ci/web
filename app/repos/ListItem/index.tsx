'use client'

import ListItem from 'app/common/ListItem'
import { RepositoryProps } from 'types/Repositories'
import API, { useAPI } from 'utils/api'

type ListItemProps = Pick<
  RepositoryProps,
  'description' | 'issues' | 'coverage' | 'name'
> & { default_branch?: string }

const RepositoryListItem = ({
  description,
  issues,
  coverage,
  name,
  default_branch,
}: ListItemProps) => {
  const { result } = useAPI(API.shared.repositoryGraphs, {
    repositoryName: name,
  })

  return (
    <div>
      <ListItem
        description={description}
        href={
          default_branch
            ? `/repos/${name}/branches/${default_branch}`
            : `/repos/${name}`
        }
        stats={{
          issues: { value: issues, data: result?.issues },
          coverage: { value: coverage, data: result?.coverage },
        }}
        title={name}
      />
    </div>
  )
}

export default RepositoryListItem
