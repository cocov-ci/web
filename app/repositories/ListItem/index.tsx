'use client'

import useSWR from 'swr'

import ListItem from 'app/common/ListItem'
import { RepositoryResponseProps } from 'types/Repositories'
import { StatsResponseProps } from 'types/Stats'

const RepositoryListItem = ({
  description,
  issues,
  coverage,
  name,
}: Pick<
  RepositoryResponseProps,
  'description' | 'issues' | 'coverage' | 'name'
>) => {
  const { data } = useSWR<StatsResponseProps>(
    `/api/repositories/${name}/graphs`,
  )

  return (
    <div>
      <ListItem
        description={description}
        href={`/repositories/${name}`}
        stats={{
          issues: { value: issues, data: data?.issues },
          coverage: { value: coverage, data: data?.coverage },
        }}
        title={name}
      />
    </div>
  )
}

export default RepositoryListItem
