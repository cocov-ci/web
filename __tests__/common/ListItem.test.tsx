import { render } from '@testing-library/react'

import ListItem from 'app/common/ListItem'

import '@testing-library/jest-dom'

describe('common/ListItem', () => {
  const ListItemComponent = () => (
    <ListItem
      description="description here..."
      stats={{
        issues: undefined,
        coverage: undefined,
      }}
      title="title here..."
    />
  )

  const { container } = render(<ListItemComponent />)

  it('renders ListItem without stats info', () => {
    expect(container.querySelector('span.backgroundText')).toHaveTextContent(
      'title here...',
    )
    expect(container.querySelector('.info p.title')).toHaveTextContent(
      'title here...',
    )
    expect(container.querySelector('.info p.description')).toHaveTextContent(
      'description here...',
    )
    expect(container.querySelectorAll('.stats p.value')[0]).toHaveTextContent(
      '—',
    )
    expect(container.querySelectorAll('.stats p.value')[1]).toHaveTextContent(
      '—',
    )
  })

  it('renders ListItem snapshots', () => {
    expect(container).toMatchSnapshot()
  })
})
