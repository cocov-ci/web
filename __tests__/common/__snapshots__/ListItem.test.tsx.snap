// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/ListItem renders ListItem snapshots 1`] = `<div />`;
